pkg/kapp/config/conf.go

```diff
@@ -31,7 +31,11 @@ func NewConfFromResources(resources []ctlres.Resource) ([]ctlres.Resource, Conf,
 		}
 	}
 
-	return rsWithoutConfigs, Conf{configs}, nil
+	cfg := Conf{configs}
+
+	ctlres.SetGlobalWaitingRules(cfg.WaitRuleMods())
```

Comment:
global variables are no-go typically, but i can take care of propagating waiting rules myself once PR is merged.